motionplan/ik/nloptInverseKinematics.go

```diff
@@ -126,11 +160,16 @@ func (ik *NloptIK) Solve(ctx context.Context,
 
 		if len(gradient) > 0 {
 			for i := range gradient {
-				x[i] += ik.jump
-				inputs = referenceframe.FloatsToInputs(x)
+				flip := false
+				inputs[i].Value += jump[i]
+				ub := ik.upperBound[i]
+				if inputs[i].Value >= ub {
+					flip = true
+					inputs[i].Value = ub - jump[i]
```

Comment:
This won't actually measure a distance of jump though.  You should go back to subtracting 2*jump from the original I think